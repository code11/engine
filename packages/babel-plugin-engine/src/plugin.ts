import * as Babel from "@babel/core";
import { VariableDeclaratorVisitor } from "./visitors";

export type PluginConfig = {
  viewLibrary: string;
};

type pluginType = (babel: typeof Babel, state: PluginConfig) => {};

export const plugin: pluginType = (babel, state) => {
  return {
    name: "babel-plugin-engine",
    visitor: {
      VariableDeclarator: VariableDeclaratorVisitor(babel, state),
    },
  };
};
