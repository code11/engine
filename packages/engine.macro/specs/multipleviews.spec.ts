import pluginTester from "babel-plugin-tester";
import prettier from "prettier";
import plugin from "babel-plugin-macros";

// const macroFile = "'./build/macro/index.macro'";
const macroFile = "'@c11/engine.macro'";
const config = require("./babelConfig.json");

pluginTester({
  plugin,
  babelOptions: { filename: __filename, ...config },
  formatResult: (result: any) => {
    return prettier.format(result, {
      parser: "babel",
    });
  },
  tests: {
    "should keep Constants": {
      only: true,
      code: `
        import { view } from ${macroFile}
        const a: view = ({
          foo = get.foo
        }) => {  }
        const b: view = ({
          foo = get.foo
        }) => {  }
      `,
      snapshot: true,
    },
  },
});
