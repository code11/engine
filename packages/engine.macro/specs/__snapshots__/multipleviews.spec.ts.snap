// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin should keep Constants: should keep Constants 1`] = `

import { view } from '@c11/engine.macro'
const a: view = ({
  foo = get.foo
}) => {  }
const b: view = ({
  foo = get.foo
}) => {  }

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _engineView = require("@c11/engine-view");

var a = (0, _engineView.view)({
  props: {
    type: "STRUCT",
    value: {
      foo: {
        type: "GET",
        path: [
          {
            type: "CONST",
            value: "foo",
          },
        ],
      },
    },
    meta: {},
  },
  fn: function fn(_ref) {
    var foo = _ref.foo;
  },
});
var b = (0, _engineView.view)({
  props: {
    type: "STRUCT",
    value: {
      foo: {
        type: "GET",
        path: [
          {
            type: "CONST",
            value: "foo",
          },
        ],
      },
    },
    meta: {},
  },
  fn: function fn(_ref2) {
    var foo = _ref2.foo;
  },
});


`;
